packages/frontend/src/app/shared/language.service.ts

```diff
@@ -0,0 +1,49 @@
+import {Injectable} from '@angular/core';
+import {TranslateService} from '@ngx-translate/core';
+import {BehaviorSubject} from 'rxjs';
+
+import {availableLanguagesCodes, AvailableLanguageToCode} from 'src/environments/available-languages-codes';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LanguageService {
+
+  constructor(private translateService: TranslateService) {
+    translateService.onLangChange.subscribe((event) => {
+      this.currentLanguage.next(event.lang);
+    });
+  }
+
+  private defaultLanguageCode: AvailableLanguageToCode = AvailableLanguageToCode.Polish;
+
+  public currentLanguage: BehaviorSubject<AvailableLanguageToCode> = new BehaviorSubject<AvailableLanguageToCode>(this.defaultLanguageCode);
```

Comment:
It's good practice to make Subjects as private field and make public variable currentLanguage$ = this.currentLanguage.asObservable()

Reply:
80a1115